File: javascript/src/sliding-window/minimum-size-subarray-sum.js
import { parseAssignments } from "../shared/io.js";
export const implemented = true;
export function solve(input) {
  const { target, nums } = parseAssignments(input);
  let left = 0, total = 0, best = Infinity;
  for (let right = 0; right < nums.length; right++) {
    total += nums[right];
    while (total >= target) {
      best = Math.min(best, right - left + 1);
      total -= nums[left];
      left++;
    }
  }
  return best === Infinity ? 0 : best;
}

File: java/src/main/java/algolens/slidingwindow/MinimumSizeSubarraySum.java
package algolens.slidingwindow;

public class MinimumSizeSubarraySum {
    public int minSubArrayLen(int target, int[] nums) {
        int l = 0, sum = 0, best = Integer.MAX_VALUE;
        for (int r = 0; r < nums.length; r++) {
            sum += nums[r];
            while (sum >= target) {
                best = Math.min(best, r - l + 1);
                sum -= nums[l];
                l++;
            }
        }
        return best == Integer.MAX_VALUE ? 0 : best;
    }
}
